tonic-build/src/lib.rs

```diff
@@ -137,7 +140,7 @@ pub trait Method {
     /// Identifier used to generate type name.
     fn identifier(&self) -> &str;
     /// Path to the codec.
-    fn codec_path(&self) -> &str;
+    fn codec_path(&self) -> String;
```

Comment:
Yeah, I was using a borrowed state (from load()) which only lives on the stack. That state can't outlive static, so I needed to return something owned. I'm looking at your other suggestion to newtype Service and add the configuration directly. Threadlocal is not my preference or remotely the first thing I tried ;-). Thanks for your feedback!

Probably if I get that resolved then the codec path will be able to be borrowed from the struct instead of the environment, which is the root of the ownership mismatch here.